### ISSUE TITLE
Weird warning when building a release version: "Missing classes detected while running R8."

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [terl/lazysodium-android#52]
When creating a release version, I've noticed this warning :

> Missing classes detected while running R8. Please add the missing classes or apply additional keep rules that are generated in ...\app\build\outputs\mapping\release\missing_rules.txt.

When looking inside, it showed various ones:

> Missing class com.google.errorprone.annotations.Immutable (referenced from: com.google.crypto.tink.KeyTemplate and 4 other contexts)
> Missing class java.awt.Component (referenced from: long com.sun.jna.Native$AWT.getComponentID(java.lang.Object) and 3 other contexts)
> Missing class java.awt.GraphicsEnvironment (referenced from: long com.sun.jna.Native$AWT.getComponentID(java.lang.Object))
> Missing class java.awt.HeadlessException (referenced from: long com.sun.jna.Native$AWT.getComponentID(java.lang.Object))
> Missing class java.awt.Window (referenced from: long com.sun.jna.Native$AWT.getWindowID(java.awt.Window) and 2 other contexts)

I don't know about the rest, but I see here "jna" , which the current repository uses.
Should I be worried?

**Comment by Ni3verma** [comment]
hey, I am getting same warning in one of my project while building release apk. 
were you able to solve it?

**Comment by gurpreet-** [comment]
I don't think this is a cause for concern as they are warnings. It does seem like a problem with JNA. It might be worth raising an issue with them @AndroidDeveloperLB

**Comment by AndroidDeveloperLB** [comment]
I don't think I see this anymore
